### ISSUE TITLE
F523: Regression: index out-of-bounds panic on invalid parameter indices

### ISSUE THREAD

**Issue by addisoncrump** [charliermarsh/ruff#4863]
Potentially, a `.format`'d string specifies a parameter outside of the number of real parameters:

```py
"{8}".format(0, 1)
```

If a) the index specified in the template exceeds the number of parameters and b) there are unused format args, the F523 fix will panic (regression from: https://github.com/charliermarsh/ruff/pull/4837)

This was detected by #4822.

**Comment by charliermarsh** [comment]
I had to modify the snippet a bit to reproduce: `"{1} {8}".format(0, 1)`
